plutus-scb/src/Cardano/Wallet/Client.hs

```diff
@@ -10,59 +10,41 @@
 module Cardano.Wallet.Client where
 
 import           Cardano.Wallet.API        (API)
-import           Cardano.Wallet.Types      (WalletId)
-import           Control.Lens
+import           Control.Monad             (void)
 import           Control.Monad.Freer
 import           Control.Monad.Freer.Error (Error, throwError)
 import           Control.Monad.IO.Class    (MonadIO (..))
 import           Data.Proxy                (Proxy (Proxy))
-import           Ledger                    (Address, PubKey, TxOutRef, Value, pubKeyAddress)
-import           Ledger.AddressMap         (AddressMap, UtxoMap, fundsAt)
-import           Servant                   (NoContent)
+import           Ledger                    (PubKey, Value)
+import           Ledger.AddressMap         (UtxoMap)
+import           Ledger.Slot               (Slot)
+import           Ledger.Tx                 (Tx)
+import           Servant                   ((:<|>) (..))
 import           Servant.Client            (ClientEnv, ClientError, ClientM, client, runClientM)
-import           Servant.Extra             (left, right)
-import           Wallet.API                (WalletAPIError)
-import           Wallet.Effects            (NodeClientEffect, WalletEffect (..), getClientSlot, publishTx)
-import           Wallet.Emulator.Wallet    (Payment (..), PaymentArgs (..), Wallet, handleUpdatePaymentWithChange)
+import           Wallet.Effects            (Payment (..), WalletEffect (..))
 
-selectCoins :: WalletId -> Value -> ClientM ([(TxOutRef, Value)], Value)
-allocateAddress :: WalletId -> ClientM PubKey
-getWatchedAddresses :: ClientM AddressMap
-getWallets :: ClientM [Wallet]
-getOwnPubKey :: ClientM PubKey
-startWatching :: Address -> ClientM NoContent
-valueAt :: Address -> ClientM Value
-(getWallets, getOwnPubKey, getWatchedAddresses, startWatching, selectCoins, allocateAddress, valueAt) =
-    ( getWallets_
-    , getOwnPubKey_
-    , getWatchedAddresses_
-    , startWatching_
-    , selectCoins_
-    , allocateAddress_
-    , valueAt_)
+submitTxn :: Tx -> ClientM ()
+ownPublicKey :: ClientM PubKey
+updatePaymentWithChange :: (Value, Payment) -> ClientM Payment
+walletSlot :: ClientM Slot
+ownOutputs :: ClientM UtxoMap
+(submitTxn, ownPublicKey, updatePaymentWithChange, walletSlot, ownOutputs) =
+  ( fmap void submitTxn_
```

Comment:
Why throw the result away?

Reply:
The result is `()` in the `WalletEffect` and `NoContent` in the servant API so this is just translating between the two. I didn't want `NoContent` to be exposed to the users of `submitTxn`